### PR TITLE
refactor(icon-toggle): Annotate mdc-icon-toggle for closure

### LINKED ISSUE #337: Annotate mdc-icon-toggle for closure

### PR DESCRIPTION
**Pull request by lynnjepsen**
Resolves #337

### CODE REVIEW

**Comment by lynnjepsen** [comment]
This is a copy of https://github.com/material-components/material-components-web/pull/873, but with the changes I originally requested. @nckh hasn't replied to the comments for more than a week, and we generally close PRs that are non-responsive, but I didn't want to lose this valuable work before closing out the PR.

File: packages/mdc-icon-toggle/adapter.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  /**
+   * @param {string} name
+   * @param {string} value
+   */
+  setAttr(name, value) {}
+
+  /** @param {string} name */
+  rmAttr(name) {}
+
+  /** @param {CheckboxEvent} evtData */
```
Comment (acdvorak) [inline packages/mdc-icon-toggle/adapter.js]: Our [style guide](https://google.github.io/styleguide/jsguide.html#jsdoc-nullability) recommends adding an explicit nullable or non-null modifier to object type annotations:

> all types except primitives and record literals must be annotated explicitly with either ? or ! to indicate whether they are nullable or not.

File: packages/mdc-icon-toggle/foundation.js [hunk 2]
```diff
@@ -121,41 +149,85 @@ export default class MDCIconToggleFoundation extends MDCFoundation {
...
+  return !!keyboardKey.key && keyboardKey.key === 'Space' || keyboardKey.keyCode === 32;
 }
+
+
+/** @record */
+class IconToggleState {}
+
+/**
+ * The aria-label value of the icon toggle, or undefined if there is no aria-label.
+ * @type {string|undefined}
```
Comment (acdvorak) [inline packages/mdc-icon-toggle/foundation.js]: These two lines can be collapsed into one: `@export {string|undefined}`. Ditto below

File: packages/mdc-icon-toggle/index.js [hunk 3]
```diff
@@ -14,28 +14,41 @@
...
 import MDCIconToggleFoundation from './foundation';
+/* eslint-disable no-unused-vars */
+import {MDCRippleAdapter} from '@material/ripple/adapter';
+/* eslint-enable no-unused-vars */
+import {MDCRipple, MDCRippleFoundation} from '@material/ripple';
 
 export {MDCIconToggleFoundation};
 
+/**
+ * @extends MDCComponent<!MDCIconToggleFoundation>
```
Comment (acdvorak) [inline packages/mdc-icon-toggle/index.js]: `@extends {MDCComponent<!MDCIconToggleFoundation>}`

**Comment by lynnjepsen** [comment]
Addressed all comments, PTAL.

**Pull request merged**